packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts

```diff
@@ -163,6 +163,14 @@ export class SageMakerCreateTrainingJob extends sfn.TaskStateBase implements iam
       throw new Error('Must define either an algorithm name or training image URI in the algorithm specification');
     }
 
+    // validate the algorithmName if the algorithmName is defined
+    if (props.algorithmSpecification.algorithmName) {
+      const regex = /^(arn:aws[a-z\-]*:sagemaker:[a-z0-9\-]*:[0-9]{12}:[a-z\-]*\/)?([a-zA-Z0-9]([a-zA-Z0-9-]){0,62})(?<!-)$/;
+      if (!regex.test(props.algorithmSpecification.algorithmName)) {
+        throw new Error(`Value '${props.algorithmSpecification.algorithmName}' at 'algorithmName' must satisfy regular expression pattern: ${regex.source}`);
```

Comment:
A message like `Expected algorithm name to match pattern ${regex}, but got ${value}` would be more inline with the conventions used in the rest of the project.

Reply:
OK, got it.
I will change the message to `Expected algorithm name to match pattern ${regex}, but got ${value}`